### PR TITLE
gdalenhance: add docs and tests

### PR DESCRIPTION
**Pull request by dbaston**
## What does this PR do?

Continues stale PR #8732, adds docs and tests to `gdalenhance`.

## What are related issues/pull requests?

#8131, #8732

### CODE REVIEW

File: doc/source/programs/gdalenhance.rst [hunk 1]
```diff
@@ -0,0 +1,94 @@
+.. _gdalenhance:
+
+================================================================================
+gdalenhance
+================================================================================
+
+.. only:: html
+
```
Comment (rouault) [inline doc/source/programs/gdalenhance.rst:8]:
```suggestion

    .. versionadded:: 3.11.0

```

Comment (rouault) [reply doc/source/programs/gdalenhance.rst:8]: actually not, it has always been there...

**Review by rouault — APPROVED** [review]

File: autotest/utilities/test_gdalenhance.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+# Write a new image directly
+
+
+def test_gdalenhance_output_image(gdalenhance_path, tmp_path):
+
+    infile = "../gcore/data/rgbsmall.tif"
+    outfile = tmp_path / "out.tif"
+
+    out, err = gdaltest.runexternal_out_and_err(
+        f"{gdalenhance_path} -quiet -equalize -co 'COMPRESS=DEFLATE' {infile} {outfile}"
```
Comment (rouault) [inline autotest/utilities/test_gdalenhance.py:79]: to make Windows happy:
```suggestion
        f"{gdalenhance_path} -quiet -equalize -co COMPRESS=DEFLATE {infile} {outfile}"
```

File: apps/gdalenhance.cpp [hunk 3]
```diff
@@ -339,6 +339,13 @@ MAIN_START(argc, argv)
         }
     }
 
+    if (padfScaleMin == nullptr || padfScaleMax == nullptr)
```
Comment (rouault) [inline apps/gdalenhance.cpp:342]: the check "if (padfScaleMin != nullptr)" at line 362 can now be removed

File: autotest/utilities/test_gdalenhance.py [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    lines = out.strip().split("\n")
+    assert len(lines) == 3
+
+    assert lines[0].startswith("1:Band ")
+    assert lines[1].startswith("2:Band ")
+    assert lines[2].startswith("3:Band ")
+
+    lut_fname = tmp_path / "lut.txt"
+
+    with open(lut_fname, "w") as outfile:
```
Comment (rouault) [inline autotest/utilities/test_gdalenhance.py:53]: I presume something like that to please Windows:
```suggestion
    with open(lut_fname, "wb") as outfile:
```

Comment (dbaston) [reply autotest/utilities/test_gdalenhance.py:53]: It was ending up with double newlines in Windows, `outfile.write(line.strip())` seems to fix

File: autotest/utilities/test_gdalenhance.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+# Write a new image directly
+
+
+def test_gdalenhance_output_image(gdalenhance_path, tmp_path):
+
+    infile = "../gcore/data/rgbsmall.tif"
+    outfile = tmp_path / "out.tif"
+
+    out, err = gdaltest.runexternal_out_and_err(
+        f"{gdalenhance_path} -quiet -equalize -co 'COMPRESS=DEFLATE' {infile} {outfile}"
```
Comment (rouault) [inline autotest/utilities/test_gdalenhance.py:79]: single-quotes have reappeared:
```suggestion
        f"{gdalenhance_path} -quiet -equalize -co COMPRESS=DEFLATE {infile} {outfile}"
```

Comment (dbaston) [reply autotest/utilities/test_gdalenhance.py]: fixed, again :disappointed:

**Pull request merged**